examples/6field-simple/elm_6f.cxx

```diff
@@ -12,1677 +12,1682 @@
-                     BoutReal n0_center, BoutReal n0_bottom_x) {
-  Field2D result;
-  result.allocate();
-
-  BoutReal Grid_NX, Grid_NXlimit; // the grid number on x, and the
-  BoutReal Jysep;
-  mesh->get(Grid_NX, "nx");
-  mesh->get(Jysep, "jyseps1_1");
-  Grid_NXlimit = n0_bottom_x * Grid_NX;
-  output.write("Jysep1_1 = {:d}   Grid number = {:e}\n", int(Jysep), Grid_NX);
-
-  if (Jysep > 0.) { // for single null geometry
-
-    BoutReal Jxsep, Jysep2;
-    mesh->get(Jxsep, "ixseps1");
-    mesh->get(Jysep2, "jyseps2_2");
-
-    for (int jx = 0; jx < mesh->LocalNx; jx++) {
-      BoutReal mgx = mesh->GlobalX(jx);
-      BoutReal xgrid_num = (Jxsep + 1.) / Grid_NX;
-      // output.write("mgx = {:e} xgrid_num = {:e}\n", mgx);
-      for (int jy = 0; jy < mesh->LocalNy; jy++) {
-        int globaly = mesh->getGlobalYIndex(jy);
-        // output.write("local y = {:d};   global y: {:d}\n", jy, globaly);
-        if (mgx > xgrid_num || (globaly <= int(Jysep) - 2) || (globaly > int(Jysep2) + 2))
+    return result;
+  }
+
+  Field2D N0tanh(BoutReal n0_height, BoutReal n0_ave, BoutReal n0_width,
+                 BoutReal n0_center, BoutReal n0_bottom_x) {
+    Field2D result;
+    result.allocate();
+
+    BoutReal Grid_NX, Grid_NXlimit; // the grid number on x, and the
+    BoutReal Jysep;
+    mesh->get(Grid_NX, "nx");
+    mesh->get(Jysep, "jyseps1_1");
+    Grid_NXlimit = n0_bottom_x * Grid_NX;
+    output.write("Jysep1_1 = {:d}   Grid number = {:e}\n", int(Jysep), Grid_NX);
+
+    if (Jysep > 0.) { // for single null geometry
+
+      BoutReal Jxsep, Jysep2;
+      mesh->get(Jxsep, "ixseps1");
+      mesh->get(Jysep2, "jyseps2_2");
+
+      for (int jx = 0; jx < mesh->LocalNx; jx++) {
+        BoutReal mgx = mesh->GlobalX(jx);
+        BoutReal xgrid_num = (Jxsep + 1.) / Grid_NX;
+        // output.write("mgx = {:e} xgrid_num = {:e}\n", mgx);
+        for (int jy = 0; jy < mesh->LocalNy; jy++) {
+          int globaly = mesh->getGlobalYIndex(jy);
+          // output.write("local y = {:d};   global y: {:d}\n", jy, globaly);
+          if (mgx > xgrid_num || (globaly <= int(Jysep) - 2)
+              || (globaly > int(Jysep2) + 2))
+            mgx = xgrid_num;
+          BoutReal rlx = mgx - n0_center;
+          BoutReal temp = exp(rlx / n0_width);
+          BoutReal dampr = ((temp - 1.0 / temp) / (temp + 1.0 / temp));
+          result(jx, jy) = 0.5 * (1.0 - dampr) * n0_height + n0_ave;
+        }
+      }
+    } else { // circular geometry
+      for (int jx = 0; jx < mesh->LocalNx; jx++) {
+        BoutReal mgx = mesh->GlobalX(jx);
+        BoutReal xgrid_num = Grid_NXlimit / Grid_NX;
+        if (mgx > xgrid_num)
           mgx = xgrid_num;
         BoutReal rlx = mgx - n0_center;
         BoutReal temp = exp(rlx / n0_width);
         BoutReal dampr = ((temp - 1.0 / temp) / (temp + 1.0 / temp));
-        result(jx, jy) = 0.5 * (1.0 - dampr) * n0_height + n0_ave;
+        for (int jy = 0; jy < mesh->LocalNy; jy++)
```

Comment:
warning: statement should be inside braces [readability-braces-around-statements]

```cpp
for (int jy = 0; jy < mesh->LocalNy; jy++)
                                          ^
                                           {
```